packages/open-api-gateway/src/project/open-api-gateway-ts-project.ts

```diff
@@ -0,0 +1,243 @@
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ******************************************************************************************************************** */
+
+import * as path from "path";
+import { SampleDir, TextFile, YamlFile } from "projen";
+import { NodePackageManager } from "projen/lib/javascript";
+import {
+  TypeScriptProject,
+  TypeScriptProjectOptions,
+} from "projen/lib/typescript";
+import { exec } from "projen/lib/util";
+import { GeneratedTypescriptClientProject } from "./codegen/generated-typescript-client-project";
+import { ClientLanguage } from "./languages";
+import {
+  getTypescriptSampleSource,
+  getTypescriptSampleTests,
+  TypescriptSampleCodeOptions,
+} from "./samples/typescript";
+import { OpenApiSpecProject } from "./spec/open-api-spec-project";
+
+const OPENAPI_GATEWAY_PDK_PACKAGE_NAME =
+  "@aws-prototyping-sdk/open-api-gateway";
+
+/**
+ * Configuration for the OpenApiGatewayTsProject
+ */
+export interface OpenApiGatewayTsProjectOptions
+  extends TypeScriptProjectOptions {
+  /**
+   * The list of languages for which clients will be generated. A typescript client will always be generated.
+   */
+  readonly clientLanguages: ClientLanguage[];
+}
+
+/**
+ * Synthesizes a Typescript Project with an OpenAPI spec, generated clients, a CDK construct for deploying the API
+ * with API Gateway, and generated lambda handler wrappers for type-safe handling of requests.
+ *
+ * @pjid open-api-gateway-ts
+ */
+export class OpenApiGatewayTsProject extends TypeScriptProject {
+  /**
+   * A reference to the generated typescript client
+   */
+  public readonly generatedTypescriptClient: TypeScriptProject;
+
+  private readonly hasParent: boolean;
+  private readonly specDir: string = "spec";
+
+  constructor(options: OpenApiGatewayTsProjectOptions) {
+    super({
+      ...options,
+      sampleCode: false,
+      tsconfig: {
+        compilerOptions: {
+          // Root dir needs to include src and spec
+          rootDir: ".",
+          lib: ["dom", "es2019"],
+        },
+      },
+    });
+
+    // Allow spec to be imported, using both the source and spec directories as project roots.
+    this.tsconfig?.addInclude(`${this.specDir}/**/*.json`);
+    this.package.addField(
+      "main",
+      path.join(this.libdir, this.srcdir, "index.js")
+    );
+    this.package.addField(
+      "types",
+      path.join(this.libdir, this.srcdir, "index.d.ts")
+    );
+
+    // Set to private since this either uses workspaces or has file dependencies
+    this.package.addField("private", true);
```

Comment:
Yes - it's a requirement since we use workspaces for the dependency on the generated code in the standalone case. It might be ok not to be private for the monorepo case though...

Reply:
Ah hold on we use workspaces in the monorepo case, and yarn/npm/pnpm link with a file dependency in the local case so I'm not sure if it's publishable in either case